pkg/controllers/secrets.go

```diff
@@ -224,14 +225,24 @@ func MountSecrets(secrets []byte, mountPath string, maxReads int) (string, func(
 				utils.HandleError(err, message)
 			}
 
-			numReads++
+			if enableReadsLimit {
+				numReads++
+			}
 			utils.LogDebug("Secrets mount opened by reader")
 
 			if _, err := f.Write(secrets); err != nil {
 				// race: cleanup has already begun; no need to error
 				if errors.Is(err, fs.ErrNotExist) && fifoCleanupStarted {
 					break
 				}
+				// broken pipe occurs when reader closes pipe before writing completes (eg. with vite dev server)
+				var errno syscall.Errno
+				if errors.As(err, &errno) && errno == syscall.EPIPE {
```

Comment:
Is there a reason for not using the simpler `errors.Is`?

```suggestion
				if errors.Is(err, syscall.EPIPE) {
```

Reply:
Ahh I believe that was left over from when I was originally trying to debug this and was having trouble with errors.Is on the wrong syscall error. Fixed now.